modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/internal/ImmediateContext.kt

```diff
@@ -0,0 +1,13 @@
+package arrow.effects.internal
+
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.ContinuationInterceptor
+import kotlin.coroutines.CoroutineContext
+
+internal object ImmediateContext : ContinuationInterceptor {
```

Comment:
We can't use that immediate dispatcher since it lives in `coroutines-core`. However, I am removing this as we speak in favor of `EmptyCoroutineContext`. That's the only adjustment I am still making in this PR because `ImmediateContext` shouldn't get merged :D 

This is currently only used to start some internal code that should be blocking. See [CancelableMVar](https://github.com/arrow-kt/arrow/pull/1132/files#diff-46b3d40c91597898c01b1d177e803b25R221) or [CancelablePromise](https://github.com/arrow-kt/arrow/pull/1132/files#diff-cf09624ff9e2cce5f9f4ee032a495cf7R92).